Userland/Applications/Browser/DownloadWidget.cpp

```diff
@@ -117,17 +119,14 @@ DownloadWidget::DownloadWidget(const URL& url)
     };
 }
 
-void DownloadWidget::did_progress(Optional<u32> total_size, u32 downloaded_size)
+void DownloadWidget::did_progress(Optional<u64> total_size, u64 downloaded_size)
 {
-    m_progressbar->set_min(0);
+    int percent = 0;
     if (total_size.has_value()) {
-        int percent = roundf(((float)downloaded_size / (float)total_size.value()) * 100.0f);
+        percent = roundf(((float)downloaded_size / (float)total_size.value()) * 100.0f);
```

Comment:
Should this calculation be done with double instead? float will be inaccurate above...16 MiB? huh...

Reply:
I've changed this to use double, as suggested. I also swapped `round` for `floor`. It seemed wrong that you only need to download 5MB of a 1GB file to get to 1%.

Reply:
Note that you could also have done this with a simple `u64` calculation of `downloaded_size * 100 / total_size`, which results in the floored percentage as well. The `* 100` does limit the range a bit, but I believe that ~164 petabytes should be enough for everyone.

Reply:
That's much simpler. I've changed it to do that instead :^)